dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -0,0 +1,163 @@
+        self,
+        gpu_stats=[],
+        model_config=mock_model_config,
+    ) -> ParallelConfig:
+        node_samples = self._extract_node_resource()
+        paral_configs: Dict[str, ParallelConfig] = {}
+        for nodes in node_samples[NodeType.WORKER]:
+            for node in nodes:
+                gpu_stats = node.used_resource.gpu_stats
+                paral_config = node.paral_config
+                data_loader_config, optimizer_config = (
+                    self._generate_dataloader_config(
+                        gpu_stats, model_config, paral_config.dataloader
+                    ),
+                    self._generate_optimizer_config(),
+                )
+                paral_configs[node.name] = ParallelConfig(
+                    data_loader_config, optimizer_config
+                )
+        logger.info("paral configs: %s", paral_configs["simple_node"])
+        return paral_configs["simple_node"]
+
+    def _generate_dataloader_config(
+        self, gpu_stats, model_config, dataloader_config
+    ):
+        if gpu_stats == []:
+            return DataLoaderConfig(0, "", 0, 0, 0)
+        # Calculate the minimum remaining memory among GPUs
+        min_remain_memory = min(
+            entry["total_memory_gb"] - entry["used_memory_gb"]
+            for entry in gpu_stats
+        )
+
+        # Update dataloader configuration version
+        updated_version = dataloader_config.version + 1
+        # Extract dataloader configuration values
+        batch_size = dataloader_config.batch_size
+        logger.info("batch size: %d", batch_size)
+
+        # Extract model configuration values
+        block_size = model_config["block_size"]
+        n_layer = model_config["n_layer"]
+        n_heads = model_config["n_heads"]
+        n_embd = model_config["n_embd"]
+
+        # Calculate the memory required for intermediate activation
+        activation_memory_gb = (
+            (
+                34 * batch_size * block_size * n_embd
+                + 5 * batch_size * (block_size**2) * n_heads
+            )
+            * n_layer
+            / (1024 * 1024 * 1024)
+        )
+
+        try:
+            updated_batch_size = int(
+                batch_size
+                + batch_size * min_remain_memory / activation_memory_gb
+            )
+        except ZeroDivisionError:
+            updated_batch_size = batch_size
+
+        return DataLoaderConfig(
+            updated_version,
+            dataloader_config.dataloader_name,
+            updated_batch_size,
+            0,
+            0,
+        )
+
+    def _generate_optimizer_config(self):
+        return OptimizerConfig(5, 6)
+
+    def _extract_node_resource(self) -> Dict[str, List[List[Node]]]:
+        stats = self._stats_collector.get_runtime_stats()
+        node_used_resources: Dict[str, List[List[Node]]] = {}
+        node_used_resources[NodeType.WORKER] = []
+        simple_node = Node(node_type="worker", node_id=0)
```

Comment:
Is it better to select the node with the max used cuda memory?

`Node(node_type="worker", node_id=0)` ->`Node(node_type=NodeType.WORKER, node_id=0)` 

```
from dlrover.python.common.constants import NodeType
```

Reply:
This is a mock node for standalone mode. The detailed node selection method is in TODO list.

Reply:
Indeed, it is reasonable to select the node with the max used Cuda memory.